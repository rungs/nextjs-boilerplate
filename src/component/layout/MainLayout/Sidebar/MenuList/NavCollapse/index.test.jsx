import React from 'react'
import NavCollapse from './index'
import { render } from '@testing-library/react'
import { usePathname } from 'next/navigation'

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux')
  return {
    ...ActualReactRedux,
    useDispatch: jest.fn(),
    useSelector: state =>
      state({
        lang: {
          lang: 'th',
        },
        isOpen: [2],
        translations: {
          translations: {
            premium_quotation: {
              common: {
                autocomplete: {
                  nodata: 'nodata',
                  loading: 'loading',
                },
              },
            },
          },
        },
      }),
  }
})

jest.mock('next/navigation')
jest.mock('@mui/material', () => {
  return {
    ...jest.requireActual('@mui/material'),
    Chip: <div></div>,
  }
})

describe('Test Component NavCollapse', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('NavCollapse component snapshot', () => {
    usePathname.mockImplementation(() => {
      return 'url'
    })
    const menu = {
      id: '4',
      icon: 'icon',
      url: 'url',
      title: 'title',
      caption: 'caption',
      children: [
        {
          id: '4',
          type: 'type',
          url: 'url',
          external: 'external',
          title: 'title',
          caption: 'caption',
          icon: 'icon',
          disabled: false,
          children: [],
        },
      ],
    }
    const { asFragment } = render(<NavCollapse menu={menu} level={1} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('render NavCollapse component', () => {
    usePathname.mockImplementation(() => {
      return ''
    })
    const menu = {
      id: '6',
      icon: 'icon',
      url: 'url',
      title: 'title',
      caption: 'caption',
      children: [
        {
          id: '6',
          type: 'collapse',
          url: 'url',
          external: 'external',
          title: 'title',
          caption: 'caption',
          icon: 'icon',
          disabled: false,
          children: [
            {
              id: 1,
              url: 'url',
              external: 'external',
              title: 'title',
              caption: 'caption',
              icon: 'icon',
              disabled: false,
            },
          ],
        },
      ],
    }

    const Component = (
      <NavCollapse menu={menu} level={0} handleClick={jest.fn()} checkOpenForParent={jest.fn()} />
    )

    render(Component)

    expect(Component).toBeDefined()
  })

  it('render NavCollapse component children type is item', () => {
    usePathname.mockImplementation(() => {
      return 'url'
    })
    const menu = {
      id: '8',
      icon: 'icon',
      url: 'url',
      title: 'title',
      caption: 'caption',
      children: [
        {
          id: '8',
          type: 'item',
          url: 'url',
          external: 'external',
          title: 'title',
          caption: 'caption',
          icon: 'icon',
          disabled: false,
          children: [
            {
              id: 1,
              url: 'url',
              external: 'external',
              title: 'title',
              caption: 'caption',
              icon: 'icon',
              disabled: false,
            },
          ],
        },
      ],
    }

    const Component = (
      <NavCollapse menu={menu} level={2} handleClick={jest.fn()} checkOpenForParent={jest.fn()} />
    )

    render(Component)

    expect(Component).toBeDefined()
  })
})
