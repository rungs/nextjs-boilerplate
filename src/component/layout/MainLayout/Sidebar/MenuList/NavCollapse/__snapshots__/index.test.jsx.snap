// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component NavCollapse NavCollapse component snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters Mui-selected MuiListItemButton-root MuiListItemButton-gutters Mui-selected css-6r21tv-MuiButtonBase-root-MuiListItemButton-root"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root css-1e09ong-MuiListItemIcon-root"
    >
      <icon
        stroke-width="1.5"
        style="margin-top: auto; margin-bottom: auto;"
      />
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1jh4dp9-MuiTypography-root"
      />
      <span
        class="MuiTypography-root MuiTypography-caption MuiTypography-gutterBottom css-621s9d-MuiTypography-root"
      >
        caption
      </span>
    </div>
    <svg
      class="icon icon-tabler icon-tabler-chevron-up"
      fill="none"
      height="1rem"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
      style="margin-top: auto; margin-bottom: auto;"
      viewBox="0 0 24 24"
      width="1rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <polyline
        points="6 15 12 9 18 15"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </div>
  <div
    class="MuiCollapse-root MuiCollapse-vertical css-1fkrdiv-MuiCollapse-root"
    style="min-height: 0px; height: 0px; transition-duration: 0ms;"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiList-root css-17s7fyi-MuiList-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-b71oe7-MuiTypography-root"
          >
            Menu Items Error
          </h6>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
